website/src/views/PageStateHandler.ts

```diff
@@ -305,3 +282,162 @@ export class GenericCompareVariantsStateHandler extends CompareVariantsStateHand
+                    ...baselineFilter,
+                    ...variantFilter.lineages,
+                    ...variantFilter.mutations,
+                },
+                displayName: this.toDisplayName(variantFilter),
+            };
+        });
+    }
+
+    public getEmptyVariantFilterConfig(): VariantFilterConfig {
+        return this.toLineageAndMutationFilterConfig({
+            lineages: {},
+            mutations: {},
+        });
+    }
+
+    public toVariantFilterConfigs(pageState: CompareVariantsData): Map<Id, VariantFilterConfig> {
+        return new Map<Id, VariantFilterConfig>(
+            Array.from(pageState.variants, ([key, variant]) => [key, this.toLineageAndMutationFilterConfig(variant)]),
+        );
+    }
+
+    private toDisplayName(variantFilter: VariantFilter) {
+        return [
+            ...Object.values(variantFilter.lineages),
+            ...(variantFilter.mutations.nucleotideMutations ?? []),
+            ...(variantFilter.mutations.aminoAcidMutations ?? []),
+            ...(variantFilter.mutations.nucleotideInsertions ?? []),
+            ...(variantFilter.mutations.aminoAcidInsertions ?? []),
+        ].join(' + ');
+    }
+
+    private writeVariantsDataToSearchParams(searchOfFilter: URLSearchParams, filter: VariantFilter): void {
+        setSearchFromLapisVariantQuery(searchOfFilter, filter, getLineageFilterFields(this.constants.lineageFilters));
+    }
+
+    private getFilter(filterParams: Map<string, string>): VariantFilter {
+        return {
+            ...getLapisVariantQuery(filterParams, getLineageFilterFields(this.constants.lineageFilters)),
+        };
+    }
+
+    private toLineageAndMutationFilterConfig(variant: VariantFilter) {
+        return {
+            lineageFilterConfigs: getLineageFilterConfigs(this.constants.lineageFilters, variant.lineages),
+            mutationFilterConfig: variant.mutations,
+        };
+    }
+}
+
+function decodeFiltersFromSearch(search: URLSearchParams, handleNonMatchingKey: (key: string) => void = () => {}) {
+    const filterMap = new Map<Id, Map<string, string>>();
+
+    for (const [key, value] of search) {
+        const keySplit = key.split($);
+        if (keySplit.length !== 2) {
+            handleNonMatchingKey(key);
+            continue;
+        }
+        const id = Number.parseInt(keySplit[1], 10);
+        if (Number.isNaN(id)) {
+            continue;
+        }
+        if (!filterMap.has(id)) {
+            filterMap.set(id, new Map<string, string>());
+        }
+        const filter = filterMap.get(id)!;
+        filter.set(keySplit[0], value);
+    }
+    return filterMap;
+}
+
+function encodeMultipleFiltersToUrlSearchParam(filters: Map<Id, Map<string, string>>) {
+    const search = new URLSearchParams();
+    for (const [id, filter] of filters) {
+        for (const [key, value] of filter) {
+            search.append(`${key}${$}${id}`, value);
+        }
+    }
+    return search;
+}
+
+type NamedLapisFilter = {
```

Comment:
We should export that from the components, too (not a task for now, but to keep it in mind)

Reply:
I added https://github.com/GenSpectrum/dashboard-components/issues/556 to keep track of it.

Reply:
We had already a ticket for that: https://github.com/GenSpectrum/dashboard-components/issues/248. I've closed the new one.